src/main/java/bio/terra/service/filedata/azure/tables/TableServiceClientUtils.java

```diff
@@ -50,12 +56,38 @@ public static boolean tableExists(TableServiceClient tableServiceClient, String
   }
 
   public static List<TableEntity> batchRetrieveFiles(
-      TableServiceClient tableServiceClient, String tableName, List<String> fileIdArray) {
+      TableServiceClient tableServiceClient, List<String> fileIdArray) {
     String filter =
         fileIdArray.stream()
             // maybe wrap or cause in parenthesis
             .map(refId -> String.format("fileId eq '%s'", refId))
             .collect(Collectors.joining(" or "));
-    return filterTable(tableServiceClient, tableName, filter);
+    return filterTable(tableServiceClient, DATASET_TABLE.toTableName(), filter);
+  }
+
+  /**
+   * Allows us to check that there is a unique entry for a given storage table and parameters passed
+   * through 'ListEntitiesOptions'
+   *
+   * @param tableServiceClient - client to use to query storage tables
+   * @param tableName - storage table to check for entry
+   * @param options - filter entities returned in list
+   * @return
+   */
+  public static boolean tableHasSingleEntry(
+      TableServiceClient tableServiceClient, String tableName, ListEntitiesOptions options) {
+    if (options == null) {
+      options = new ListEntitiesOptions();
+    }
+    if (tableHasEntries(tableServiceClient, tableName, options)) {
+      TableClient tableClient = tableServiceClient.getTableClient(tableName);
+      PagedIterable<TableEntity> entities = tableClient.listEntities(options, null, null);
```

Comment:
You could also `setTop(2)` here to limit the number of entities returned by the query

Reply:
Great point! I'll make that change.